packages/ui/src/views/repo/components/branch-selector/branch-selector-dropdown.tsx

```diff
@@ -33,7 +33,7 @@ export const getBranchSelectorLabels = (t: TFunction) => ({
 })
 
 export interface BranchSelectorDropdownProps {
-  selectedBranch: BranchSelectorListItem
+  selectedBranch?: BranchSelectorListItem
```

Comment:
does this need to be optional?

Reply:
Yes, if there's no selected branch, only then we update it in another page or else we retain the state.